src/adapters/intention.js

```diff
@@ -0,0 +1,21 @@
+import CategoryService from 'src/adapters/category_service';
+
+export default class Intention {
+  constructor({ filter, description }) {
+    this.category = CategoryService.getCategoryByName(filter.category);
+    this.name = this.category ? this.category.getInputValue() : '';
+    this.bbox = filter.bbox;
+    this.place = description.place;
+  }
+
+  toQueryString = () => {
```

Comment:
perfect time to reuse `buildQueryString` from `url_utils.js`. There is not much added value in this case though, but it's a start. However, please note `toQueryString` already return a `?` if there is at least one param

Reply:
we ca also extends `buildQueryString` to ignore undefined values btw

Reply:
I did that :) https://github.com/QwantResearch/erdapfel/pull/606/commits/b690f19579f5ddfd73e16359639fc7c72f94f514